dev_tools/nyaml2nxdl/nyaml2nxdl_backward_tools.py

```diff
@@ -0,0 +1,950 @@
+            tag = remove_namespace_from_tag(child.tag)
+            child_attrs = child.attrib
+            # taking care of index and value attributes
+            if tag == "dim":
+                # taking care of index and value in format [[index, value]]
+                index = child_attrs.get("index", "")
+                value = child_attrs.get("value", "")
+                dim_index_value = f"{dim_index_value}[{index}, {value}], "
+                if "index" in child_attrs:
+                    del child_attrs["index"]
+                if "value" in child_attrs:
+                    del child_attrs["value"]
+
+                # Taking care of doc comes as child of dim
+                for cchild in list(child):
+                    ttag = cchild.tag.split("}", 1)[1]
+                    if ttag == ("doc"):
+                        if ttag not in dim_other_parts:
+                            dim_other_parts[ttag] = []
+                        text = cchild.text
+                        dim_other_parts[ttag].append(text.strip())
+                        child.remove(cchild)
+                        continue
+                # taking care of other attributes except index and value
+                for attr, value in child_attrs.items():
+                    if attr in possible_dim_attrs:
+                        if attr not in dim_other_parts:
+                            dim_other_parts[attr] = []
+                        dim_other_parts[attr].append(value)
+            if tag == CMNT_TAG and self.include_comment:
+                # Store and remove node so that comment nodes from dim node so
+                # that it does not call in xmlparser function
+                dim_cmnt_node.append(child)
+                node.remove(child)
+
+        # All 'dim' element comments on top of 'dim' yaml key
+        if dim_cmnt_node:
+            for ch_nd in dim_cmnt_node:
+                self.handle_comment(depth + 1, ch_nd, file_out)
+        # index and value attributes of dim elements
+        indent = (depth + 1) * DEPTH_SIZE
+        value = dim_index_value[:-2] or ""
+        file_out.write(f"{indent}dim: [{value}]\n")
+
+        # Write the attributes, except index and value, and doc of dim as child of dim_parameter.
+        # But the doc or attributes for each dim come inside list according to the order of dim.
+        if dim_other_parts:
+            indent = (depth + 1) * DEPTH_SIZE
+            file_out.write(f"{indent}dim_parameters:\n")
+            # depth = depth + 2 dim_paramerter has child such as doc of dim
+            indent = (depth + 2) * DEPTH_SIZE
+            for key, value in dim_other_parts.items():
+                if key == "doc":
+                    value = self.handle_not_root_level_doc(
+                        depth + 2, str(value), key, file_out
+                    )
+                else:
+                    # Increase depth size inside handle_map...() for writting text with one
```

Comment:
"... writing ...". You should search and replace for these regular misspellings.